go-controller/pkg/ovn/network_controller_info_pods.go

```diff
@@ -0,0 +1,694 @@
+	// is created by the node watch.
+	var subnets []*net.IPNet
+	if err := wait.PollImmediate(30*time.Millisecond, 30*time.Second, func() (bool, error) {
+		subnets = nci.lsManager.GetSwitchSubnets(switchName)
+		return subnets != nil, nil
+	}); err != nil {
+		return fmt.Errorf("timed out waiting for logical switch %q subnet: %v", switchName, err)
+	}
+	return nil
+}
+
+func (nci *NetworkControllerInfo) addRoutesGatewayIP(pod *kapi.Pod, podAnnotation *util.PodAnnotation,
+	nodeSubnets []*net.IPNet) error {
+	// if there are other network attachments for the pod, then check if those network-attachment's
+	// annotation has default-route key. If present, then we need to skip adding default route for
+	// OVN interface
+	networks, err := util.GetK8sPodAllNetworks(pod)
+	if err != nil {
+		return fmt.Errorf("error while getting network attachment definition for [%s/%s]: %v",
+			pod.Namespace, pod.Name, err)
+	}
+	otherDefaultRouteV4 := false
+	otherDefaultRouteV6 := false
+	for _, network := range networks {
+		for _, gatewayRequest := range network.GatewayRequest {
+			if utilnet.IsIPv6(gatewayRequest) {
+				otherDefaultRouteV6 = true
+			} else {
+				otherDefaultRouteV4 = true
+			}
+		}
+	}
+
+	for _, podIfAddr := range podAnnotation.IPs {
+		isIPv6 := utilnet.IsIPv6CIDR(podIfAddr)
+		nodeSubnet, err := util.MatchIPNetFamily(isIPv6, nodeSubnets)
+		if err != nil {
+			return err
+		}
+
+		gatewayIPnet := util.GetNodeGatewayIfAddr(nodeSubnet)
+
+		otherDefaultRoute := otherDefaultRouteV4
+		if isIPv6 {
+			otherDefaultRoute = otherDefaultRouteV6
+		}
+		var gatewayIP net.IP
+		if otherDefaultRoute {
+			for _, clusterSubnet := range config.Default.ClusterSubnets {
+				if isIPv6 == utilnet.IsIPv6CIDR(clusterSubnet.CIDR) {
+					podAnnotation.Routes = append(podAnnotation.Routes, util.PodRoute{
+						Dest:    clusterSubnet.CIDR,
+						NextHop: gatewayIPnet.IP,
+					})
+				}
+			}
+			for _, serviceSubnet := range config.Kubernetes.ServiceCIDRs {
+				if isIPv6 == utilnet.IsIPv6CIDR(serviceSubnet) {
+					podAnnotation.Routes = append(podAnnotation.Routes, util.PodRoute{
+						Dest:    serviceSubnet,
+						NextHop: gatewayIPnet.IP,
+					})
+				}
+			}
+		} else {
+			gatewayIP = gatewayIPnet.IP
+		}
+
+		if gatewayIP != nil {
+			podAnnotation.Gateways = append(podAnnotation.Gateways, gatewayIP)
+		}
+	}
+	return nil
+}
+
+// podExpectedInLogicalCache returns true if pod should be added to oc.logicalPortCache.
+// For some pods, like hostNetwork pods, overlay node pods, or completed pods waiting for them to be added
+// to oc.logicalPortCache will never succeed.
+func (nci *NetworkControllerInfo) podExpectedInLogicalCache(pod *kapi.Pod) bool {
+	switchName := pod.Spec.NodeName
+	return util.PodWantsNetwork(pod) && !nci.lsManager.IsNonHostSubnetSwitch(switchName) && !util.PodCompleted(pod)
+}
+
+func (nci *NetworkControllerInfo) addPodLogicalPort(pod *kapi.Pod,
```

Comment:
A bit confusing that the name `addPodLogicalPort` is so similar to `addLogicalPort`. Would it work better to have `addLogicalPortToNetwork' or similar?